test/integration/ipamd/ipamd_suite_test.go

```diff
@@ -43,22 +51,54 @@ var _ = BeforeSuite(func() {
 	numOfNodes = len(nodeList.Items)
 	Expect(numOfNodes).Should(BeNumerically(">", 1))
 
-	// Nominate the first node as the primary node
-	primaryNode = nodeList.Items[0]
+	// Nominate the first node as the 'primary' node and force coredns deployment on this
+	By("adding nodeSelector in coredns deployment to be scheduled on single node")
+	instanceID := k8sUtils.GetInstanceIDFromNode(nodeList.Items[0])
+	primaryInstance, err = f.CloudServices.EC2().DescribeInstance(instanceID)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Add nodeSelector label to coredns deployment so coredns pods are scheduled on 'primary' node
+	By("getting node with no pods scheduled to run tests")
+	coreDNSDeployment, err := f.K8sResourceManagers.DeploymentManager().GetDeployment(CoreDNSDeploymentName,
+		CoreDNSDeploymentNameSpace)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Copy the deployment to restore later
+	coreDNSDeploymentCopy = coreDNSDeployment.DeepCopy()
+
+	coreDNSDeployment.Spec.Template.Spec.NodeSelector = map[string]string{
+		"kubernetes.io/hostname": *primaryInstance.PrivateDnsName,
```

Comment:
Should we also set coredns replica to 1?

Reply:
What difference will it make?

Reply:
Just wondering since it's not best practice to have both replicas on same node generally. 

I think it is okay in this case, not much of a difference.